nilearn/regions/tests/test_region_extractor.py

```diff
@@ -51,10 +51,15 @@ def test_threshold_maps_ratio():
     # smoke test for function _threshold_maps_ratio with randomly
     # generated maps
 
-    # make sure that n_regions (4th dimension) are kept same even
-    # in thresholded image
+    # Check that the input maps are not changed
     maps, _ = generate_maps((6, 8, 10), n_regions=3)
+    maps.get_data()[:3] = 100
+    maps_data = maps.get_data().copy()
     thr_maps = _threshold_maps_ratio(maps, threshold=1.0)
+    np.testing.assert_array_equal(maps.get_data(), maps_data)
```

Comment:
Have you checked that this test fails with the current codebase ?
Looks good otherwise.

Reply:
yes, and this is why I added the line `maps.get_data()[:3] = 100`. Otherwise actually the thresholded maps are exactly equal to the maps, so the test was passing with the current codebase.

Reply:
what I don't understand is that I was assuming that _safe_get_data is precisely supposed to avoid such side effects but this doesn't work ! I didn't go deep into this but might be interesting to have a look at _safe_get_data behaviour

Reply:
may be I am missing something here

Reply:
No, _safe_get_data avoids that the in-cache vs out-of-cache status of an image changes. But it doesn't make a copy.

Reply:
Ah, Ok ! Thank you for the explanation